pkg/util/fileconfig.go

```diff
@@ -0,0 +1,257 @@
+}
+
+// ClusterConfigJSONv1 strongly typed JSON spec for cluster-config above
+type ClusterConfigJSONv1 struct {
+    Version     int              `json:"version"`
+    ClusterConf *ClusterConfigv1 `json:"cluster-config"`
+}
+
+// Constants and enum for constructPath operation
+type pathType int
+
+const (
+    clusterConfig pathType = 0
+    pubSubject    pathType = 1
+    pubCreds      pathType = 2
+    provSubject   pathType = 3
+    provCreds     pathType = 4
+)
+
+const (
+    fNamePrefix      = "ceph-cluster"
+    fNameSep         = "-"
+    fNamePubPrefix   = "publish-secret"
+    fNameProvPrefix  = "provisioner-secret"
+    fNameCephConfig  = "cluster-config"
+    fNamePubSubject  = "subjectid"
+    fNameProvSubject = "subjectid"
+    fNamePubCred     = "credentials"
+    fNameProvCred    = "credentials"
+)
+
+// constructPath constructs well defined paths based on the type of config
+// file that needs to be accessed.
+func (pType pathType) constructPath(basepath string, fsid string) (filePath string, noerr error) {
+    if fsid == "" || basepath == "" {
+        return "", fmt.Errorf("missing/empty fsid (%s) or basepath (%s) for config files", fsid, basepath)
+    }
+
+    switch pType {
+    case clusterConfig:
+        filePath = basepath + "/" + fNamePrefix + fNameSep + fsid +
+            "/" + fNameCephConfig
+    case pubSubject:
+        filePath = basepath + "/" + fNamePrefix + fNameSep + fsid +
+            fNameSep + fNamePubPrefix + "/" + fNamePubSubject
+    case pubCreds:
+        filePath = basepath + "/" + fNamePrefix + fNameSep + fsid +
+            fNameSep + fNamePubPrefix + "/" + fNamePubCred
+    case provSubject:
+        filePath = basepath + "/" + fNamePrefix + fNameSep + fsid +
+            fNameSep + fNameProvPrefix + "/" + fNameProvSubject
+    case provCreds:
+        filePath = basepath + "/" + fNamePrefix + fNameSep + fsid +
+            fNameSep + fNameProvPrefix + "/" + fNameProvCred
+    default:
+        return "", fmt.Errorf("invalid path type (%d) specified", pType)
+    }
+
+    return
+}
+
+// GetMons returns a comma separated MON list, that is read in from the config
+// files, based on the passed in fsid
+func (fc *FileConfig) GetMons(fsid string) (string, error) {
+    fPath, err := clusterConfig.constructPath(fc.BasePath, fsid)
+    if err != nil {
+        return "", err
+    }
+
+    // #nosec
+    contentRaw, err := ioutil.ReadFile(fPath)
+    if err != nil {
+        return "", err
+    }
+
+    var cephConfig ClusterConfigJSONv1
+
+    err = json.Unmarshal(contentRaw, &cephConfig)
+    if err != nil {
+        return "", err
+    }
+
+    if cephConfig.ClusterConf.ClusterFsID != fsid {
+        return "", fmt.Errorf("mismatching Ceph cluster fsid (%s) in file, passed in (%s)", cephConfig.ClusterConf.ClusterFsID, fsid)
+    }
+
+    if len(cephConfig.ClusterConf.Monitors) == 0 {
+        return "", fmt.Errorf("monitor list empty in configuration file")
```

Comment:
use `errors.New` instead of `fmt.Errorf ` if we don't need formatting

Reply:
Done.